src/redux/data.ts

```diff
@@ -363,38 +800,64 @@ export const dataUpdateAsset = assetData => (dispatch, getState) => {
   };
   dispatch({
     payload: updatedAssetsData,
-    type: DATA_UPDATE_ACCOUNT_ASSETS_DATA,
+    type: DATA_LOAD_ACCOUNT_ASSETS_DATA_SUCCESS,
   });
   saveAccountAssetsData(updatedAssetsData, accountAddress, network);
 };
 
-export const dataUpdateAssets = assetsData => (dispatch, getState) => {
+/**
+ * Replaces the account asset data in state and saves to account local storage.
+ *
+ * @param assetsData The new asset data.
+ */
+export const dataUpdateAssets = (assetsData: {
+  [uniqueId: string]: ParsedAddressAsset;
+}) => (
+  dispatch: Dispatch<DataLoadAccountAssetsDataSuccessAction>,
+  getState: AppGetState
+) => {
   const { accountAddress, network } = getState().settings;
   if (!isEmpty(assetsData)) {
     saveAccountAssetsData(assetsData, accountAddress, network);
     // Change the state since the account isn't empty anymore
     saveAccountEmptyState(false, accountAddress, network);
     dispatch({
       payload: assetsData,
-      type: DATA_UPDATE_ACCOUNT_ASSETS_DATA,
+      type: DATA_LOAD_ACCOUNT_ASSETS_DATA_SUCCESS,
     });
   }
 };
 
-const checkMeta = message => (dispatch, getState) => {
+/**
+ * Checks whether or not metadata received from Zerion is valid.
+ *
+ * @param message The message received from Zerion.
+ */
+const checkMeta = (message: DataMessage | undefined) => (
+  dispatch: Dispatch<never>,
+  getState: AppGetState
+) => {
   const { accountAddress, nativeCurrency } = getState().settings;
   const address = message?.meta?.address;
   const currency = message?.meta?.currency;
   return (
-    isLowerCaseMatch(address, accountAddress) &&
-    isLowerCaseMatch(currency, nativeCurrency)
+    isLowerCaseMatch(address!, accountAddress) &&
+    isLowerCaseMatch(currency!, nativeCurrency)
   );
 };
 
-const checkForConfirmedSavingsActions = transactionsData => dispatch => {
+/**
+ * Checks to see if new savings are available based on incoming transaction data,
+ * and if so, updates state to request refetching savings.
+ *
+ * @param transactionsData Incoming transaction data.
+ */
+const checkForConfirmedSavingsActions = (
+  transactionsData: ZerionTransaction[]
+) => (dispatch: ThunkDispatch<AppState, unknown, never>) => {
   const foundConfirmedSavings = find(
```

Comment:
This `find` function previously wasn't imported, so I added it to the Lodash import. Is there any reason why it shouldn't be imported now? (For what it's worth, I'm not sure how this would have worked before without the import, but I might be missing something.)

Reply:
woops yeah I dunno why it doesn't complain when this happens (I've seen this happen before)